Makefile isn't working properly

**Describe the bug**
Something is broken for developers setting up a local dev environment, as described here: https://github.com/scidsg/hushline/tree/main/docs#getting-started


**To Reproduce**

- Clone the hushline repo
- `make run` to start all the containers
- `make migrate-dev`, `make migrate-prod`, and `make revision message='test'` should all work

The actually all fail:

```
user@cloak ~/code/scidsg/hushline-tmp (main) $ make migrate-dev
poetry run ./scripts/dev_migrations.py
Creating database tables
Traceback (most recent call last):
  File "/home/user/code/scidsg/hushline-tmp/./scripts/dev_migrations.py", line 15, in <module>
    main()
  File "/home/user/code/scidsg/hushline-tmp/./scripts/dev_migrations.py", line 8, in main
    with create_app().app_context():
         ^^^^^^^^^^^^
  File "/home/user/code/scidsg/hushline-tmp/hushline/__init__.py", line 32, in create_app
    db.init_app(app)
  File "/home/user/.cache/pypoetry/virtualenvs/hushline-g1GIqsd5-py3.12/lib/python3.12/site-packages/flask_sqlalchemy/extension.py", line 355, in init_app
    raise RuntimeError(
RuntimeError: Either 'SQLALCHEMY_DATABASE_URI' or 'SQLALCHEMY_BINDS' must be set.
make: *** [Makefile:23: migrate-dev] Error 1
user@cloak ~/code/scidsg/hushline-tmp (main) $ make migrate-prod
poetry run flask db upgrade
Error: Could not locate a Flask application. Use the 'flask --app' option, 'FLASK_APP' environment variable, or a 'wsgi.py' or 'app.py' file in the current directory.

Usage: flask [OPTIONS] COMMAND [ARGS]...
Try 'flask --help' for help.

Error: No such command 'db'.
make: *** [Makefile:27: migrate-prod] Error 2
user@cloak ~/code/scidsg/hushline-tmp (main) $ make revision message='test'
poetry run flask db upgrade
Error: Could not locate a Flask application. Use the 'flask --app' option, 'FLASK_APP' environment variable, or a 'wsgi.py' or 'app.py' file in the current directory.

Usage: flask [OPTIONS] COMMAND [ARGS]...
Try 'flask --help' for help.

Error: No such command 'db'.
make: *** [Makefile:27: migrate-prod] Error 2
```

**Expected behavior**
The dev database should get migrated, and it should autogenerate a migration.

--- a/Makefile
+++ b/Makefile
@@ -20,11 +20,11 @@ run-full: ## Run the app with all features enabled
 
 .PHONY: migrate-dev
 migrate-dev: ## Run dev env migrations
-	poetry run ./scripts/dev_migrations.py
+	docker compose run app poetry run ./scripts/dev_migrations.py
 
 .PHONY: migrate-prod
 migrate-prod: ## Run prod env (alembic) migrations
-	poetry run flask db upgrade
+	docker compose run app poetry run flask db upgrade
 
 .PHONY: dev-data
 dev-data: migrate-dev ## Run dev env migrations, and add dev data
@@ -48,7 +48,7 @@ revision: migrate-prod ## Create a new migration
 ifndef message
 	$(error 'message' must be set when invoking the revision target, eg `make revision message="short message"`)
 endif
-	poetry run flask db revision -m "$(message)" --autogenerate
+	docker compose run app poetry run flask db revision -m "$(message)" --autogenerate
 
 TESTS ?= ./tests/
 .PHONY: test

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,11 @@
 .DEFAULT_GOAL := help
 
+ifndef IS_DOCKER
+CMD := docker compose run --rm app
+else
+CMD :=
+endif
+
 .PHONY: help
 help: ## Print the help message
 	@awk 'BEGIN {FS = ":.*?## "} /^[0-9a-zA-Z_-]+:.*?## / {printf "\033[36m%s\033[0m : %s\n", $$1, $$2}' $(MAKEFILE_LIST) | \
@@ -20,38 +26,37 @@ run-full: ## Run the app with all features enabled
 
 .PHONY: migrate-dev
 migrate-dev: ## Run dev env migrations
-	docker compose run app poetry run ./scripts/dev_migrations.py
+	$(CMD) poetry run ./scripts/dev_migrations.py
 
 .PHONY: migrate-prod
 migrate-prod: ## Run prod env (alembic) migrations
-	docker compose run app poetry run flask db upgrade
+	$(CMD) poetry run flask db upgrade
 
 .PHONY: dev-data
 dev-data: migrate-dev ## Run dev env migrations, and add dev data
-	poetry run ./scripts/dev_data.py
+	$(CMD) poetry run ./scripts/dev_data.py
 
 .PHONY: lint
 lint: ## Lint the code
-	poetry run ruff format --check && \
-	poetry run ruff check --output-format full && \
-	poetry run mypy . && \
-	docker compose run --rm app npx prettier --check ./*.md ./docs ./.github/workflows/* ./hushline
+	$(CMD) poetry run ruff format --check && \
+	$(CMD) poetry run ruff check --output-format full && \
+	$(CMD) poetry run mypy . && \
+	$(CMD) npx prettier --check ./*.md ./docs ./.github/workflows/* ./hushline
 
 .PHONY: fix
 fix: ## Format the code
-	poetry run ruff format && \
-	poetry run ruff check --fix && \
-	docker compose run --rm app npx prettier --write ./*.md ./docs ./.github/workflows/* ./hushline
+	$(CMD) poetry run ruff format && \
+	$(CMD) poetry run ruff check --fix && \
+	$(CMD) npx prettier --write ./*.md ./docs ./.github/workflows/* ./hushline
 
 .PHONY: revision
 revision: migrate-prod ## Create a new migration
 ifndef message
 	$(error 'message' must be set when invoking the revision target, eg `make revision message="short message"`)
 endif
-	docker compose run app poetry run flask db revision -m "$(message)" --autogenerate
+	$(CMD) poetry run flask db revision -m "$(message)" --autogenerate
 
 TESTS ?= ./tests/
 .PHONY: test
 test: ## Run the test suite
-	docker compose run --rm app \
-		poetry run pytest --cov hushline --cov-report term --cov-report html -vv $(PYTEST_ADDOPTS) $(TESTS)
+	$(CMD) poetry run pytest --cov hushline --cov-report term --cov-report html -vv $(PYTEST_ADDOPTS) $(TESTS)

--- a/docker-compose.staging.yaml
+++ b/docker-compose.staging.yaml
@@ -7,6 +7,7 @@ services:
     ports:
       - 127.0.0.1:8080:8080
     environment:
+      IS_DOCKER: '1'
       FLASK_APP: hushline
       FLASK_ENV: production
       ENCRYPTION_KEY: bi5FDwhZGKfc4urLJ_ChGtIAaOPgxd3RDOhnvct10mw=

--- a/docker-compose.stripe.yaml
+++ b/docker-compose.stripe.yaml
@@ -7,6 +7,7 @@ services:
     ports:
       - 127.0.0.1:8080:8080
     environment:
+      IS_DOCKER: '1'
       FLASK_APP: hushline
       FLASK_ENV: development
       ENCRYPTION_KEY: bi5FDwhZGKfc4urLJ_ChGtIAaOPgxd3RDOhnvct10mw=

--- a/docker-compose.yaml
+++ b/docker-compose.yaml
@@ -7,6 +7,7 @@ services:
     ports:
       - 127.0.0.1:8080:8080
     environment:
+      IS_DOCKER: '1'
       FLASK_APP: hushline
       FLASK_ENV: development
       ENCRYPTION_KEY: bi5FDwhZGKfc4urLJ_ChGtIAaOPgxd3RDOhnvct10mw=

--- a/scripts/dev_data.py
+++ b/scripts/dev_data.py
@@ -12,10 +12,10 @@ from hushline.storage import S3Driver, public_store
 
 def main() -> None:
     print("Adding dev data")
-    create_app().app_context().push()
-    create_users()
-    create_tiers()
-    create_localstack_buckets()
+    with create_app().app_context():
+        create_users()
+        create_tiers()
+        create_localstack_buckets()
 
 
 def create_users() -> None:

